src/mono/mono/mini/method-to-ir.c

```diff
@@ -10081,6 +10079,16 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				is_instance = FALSE;
 			}
 
+			is_static = (il_op == MONO_CEE_LDSFLD || il_op == MONO_CEE_LDSFLDA || il_op == MONO_CEE_STSFLD);
+			if (is_static && mono_class_is_ginst (klass)) {
+				MonoGenericContext *context = &mono_class_get_generic_class (klass)->context;
+				if (context->class_inst) {
+					for (guint i = 0; i < context->class_inst->type_argc; ++i)
+						if (m_class_is_byreflike (mono_class_from_mono_type_internal (context->class_inst->type_argv [i])))
+							TYPE_LOAD_ERROR (klass);
```

Comment:
I dont' understand this check.

I think it's looking for code like this:

```
   stsfld SomeClass::SomeStaticField
```

where `SomeStaticField` is `Foo<ByRefLikeStruct>` is that right?

That doesn't seem illegal to me.

I would expect something like this type `Foo<T>` to be illegal because of the bad field.  but the good field is ok.

```
  class Foo<T> where T : allows ref struct
  {
       public static T badField;
       public static Bar<T> okField;
  }
  class Bar<T> where T: allows ref struct {
     // doesn't use T for any fields
  }
```

I would expect the type loader to mark the generic instances of `Foo` with an error.  And the JIT for `ldsfld` would just check that `klass` isn't marked with an error.

Reply:
The spec isn't as clear as it should be, sorry. @lambdageek is correct in that `public static Bar<T> okField;` is valid to load as a field.